app/src/ui/diff/diff-options.tsx

```diff
@@ -4,14 +4,13 @@ import { Octicon } from '../octicons'
 import * as OcticonSymbol from '../octicons/octicons.generated'
 import { RadioButton } from '../lib/radio-button'
 import { Popover, PopoverCaretPosition } from '../lib/popover'
-import { RepositorySectionTab } from '../../lib/app-state'
 
 interface IDiffOptionsProps {
-  readonly sourceTab: RepositorySectionTab
+  readonly isInteractiveDiff: boolean
   readonly hideWhitespaceChanges: boolean
   readonly onHideWhitespaceChangesChanged: (
     hideWhitespaceChanges: boolean
-  ) => Promise<void>
+  ) => Promise<void> | void
```

Comment:
This is odd 🤔  I think this should just return `void`, since nothing else is awaiting for this, no?
```suggestion
  ) => void
```

Reply:
Yeah. I am not sure why this is set up as promise in diff options and it required a promise to be returned. I will see if there is a cleaner way to handle it.

Reply:
Traced it back to the checkbox originating it and I can't find a reason that it was a Promise. We are not waiting on that to return before doing anything. It seems to work just fine too (reason I didn't change it was I didn't want to impact the changes/history view).